components/connectivity-adapter/internal/appregistry/appdetails/middleware.go

```diff
@@ -1,3 +1,74 @@
 package appdetails
 
-// TODO: Write middleware fetching Application by name
+import (
+	"fmt"
+	"net/http"
+
+	"github.com/gorilla/mux"
+	"github.com/kyma-incubator/compass/components/connectivity-adapter/pkg/apperrors"
+	"github.com/kyma-incubator/compass/components/connectivity-adapter/pkg/gqlcli"
+	"github.com/kyma-incubator/compass/components/connectivity-adapter/pkg/reqerror"
+	"github.com/kyma-incubator/compass/components/director/pkg/graphql"
+	"github.com/kyma-incubator/compass/tests/director/pkg/gql"
+	gcli "github.com/machinebox/graphql"
+	"github.com/pkg/errors"
+	log "github.com/sirupsen/logrus"
+)
+
+const appNameVar = "app-name"
+const nameKey = "name"
+
+//go:generate mockery -name=GraphQLRequestBuilder -output=automock -outpkg=automock -case=underscore
+type GraphQLRequestBuilder interface {
```

Comment:
why your dependency returns graphql query? IMO you should GraphQL should be hidden from middleware and GetApplicationByName (not Applications) should return concrete application